### PR TITLE
Variable watching for non-ps object

### PR DESCRIPTION
**Pull request by ericdevg**
Fixed issue #83 
Note: for COM object, we still not able to watch, refer to #77 
@AndreSayreMSFT @HuanhuanSunMSFT

### CODE REVIEW

File: PowerShellTools/DebugEngine/ScriptProperty.cs [hunk 1]
```diff
@@ -114,16 +110,19 @@ public class ScriptProperty : IDebugProperty2
 
         protected readonly ScriptDebugger _debugger;
 
-        public ScriptProperty(ScriptDebugger debugger, string name, string value, string type)
+        protected string _path { get; set; }
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/ScriptProperty.cs]: Can path ever change after initialization?  Why not protected Path{ get; private set;} with it being set inside the base(ScriptDebugger debugger, Variable var, NewPathStuffHere) constructor?

File: PowerShellTools/DebugEngine/ScriptProperty.cs [hunk 2]
```diff
@@ -114,16 +110,19 @@ public class ScriptProperty : IDebugProperty2
...
             Log.DebugFormat("{0} {1}", name, value);
             Name = name;
             Value = value;
             _debugger = debugger;
             TypeName = (type == null ? string.Empty : type);
+            _path = path;
         }
 
-        public ScriptProperty(ScriptDebugger debugger, Variable var)
+        protected ScriptProperty(ScriptDebugger debugger, Variable var)
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/ScriptProperty.cs]: Above re: constructor taking in Path

File: PowerShellTools/DebugEngine/ScriptProperty.cs [hunk 3]
```diff
@@ -114,16 +110,19 @@ public class ScriptProperty : IDebugProperty2
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/ScriptProperty.cs:108]: General Q: Do all of these properties need to be public?

File: PowerShellTools/DebugEngine/ScriptProperty.cs [hunk 4]
```diff
@@ -197,8 +196,14 @@ public int EnumChildren(enum_DEBUGPROP_INFO_FLAGS dwFields, uint dwRadix, ref Gu
 
         protected virtual IEnumerable<ScriptProperty> GetChildren()
         {
-            var props = Value.GetType().GetProperties(BindingFlags.Public | BindingFlags.Instance);
-            return props.Select(propertyInfo => ScriptPropertyFactory.MakeProperty(_debugger, new Variable(propertyInfo.Name, propertyInfo.GetValue(Value, null).ToString(), string.Empty, false, false), string.Empty));
+            string varFullName = string.IsNullOrEmpty(_path) ? Value : string.Format("{0}\\{1}", _path, Name);
+
+            Collection<Variable> propVars = _debugger.DebuggingService.GetObjectVariable(varFullName);
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/ScriptProperty.cs:200]: Any reason we can't use IEnumerable<T> here?

Comment (ASayre) [inline PowerShellTools/DebugEngine/ScriptProperty.cs:200]: [Guessing related to WCF, but wanted to confirm for my own knowledge :)]

File: PowerShellTools/DebugEngine/ScriptProperty.cs (hunk 1 above)
Comment (ericdevg) [inline PowerShellTools/DebugEngine/ScriptProperty.cs]: Sure, fixed.

File: PowerShellTools/DebugEngine/ScriptProperty.cs (hunk 4 above)
Comment (ericdevg) [inline PowerShellTools/DebugEngine/ScriptProperty.cs:200]: Correct, it won't be able to serialize IEnumerable

File: PowerShellTools/DebugEngine/ScriptProperty.cs (hunk 2 above)
Comment (ericdevg) [inline PowerShellTools/DebugEngine/ScriptProperty.cs]: Fixed by adding path, so that derived one does not need to set by its own.

**Pull request merged**